### PR TITLE
Max file size error message

### PR DESCRIPTION
**Pull request by chobeat**

### CODE REVIEW

File: test/file_error_test.exs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  alias Bonfire.Files.FileDenied
+  alias Bonfire.Files.IconUploader
+  alias Bonfire.Files.ImageUploader
+  alias Bonfire.Files.Media
+
+  @doc """
+  This macro temporarily replaces an environment variable for testing purposes.
+  Since the environment is global to the process, this is not async safe and might interfere with other tests if run in parallel.
+  """
+  defmacro with_var(app, key, value, do: expression) do
```
Comment (mayel) [inline test/file_error_test.exs]: This looks like it'd work fine, but just to point out an example of a more idiomatic approach: https://github.com/bonfire-networks/bonfire_ui_social/blob/main/test/views/social_settings_test.exs#L15

Using one of ex_unit's 4 setup functions: https://hexdocs.pm/ex_unit/ExUnit.Callbacks.html#setup/1 and on_exit: https://hexdocs.pm/ex_unit/ExUnit.Callbacks.html#on_exit/2

File: mix.exs [hunk 2]
```diff
@@ -29,7 +29,8 @@ defmodule Bonfire.Files.MixProject do
           {:image, "~> 0.37", optional: true},
           {:evision, "~> 0.1", optional: true},
           {:ex_aws_s3, "~> 2.3", optional: true},
-          {:blurhash, "~> 0.1.0", hex: :rinpatch_blurhash}
+          {:blurhash, "~> 0.1.0", hex: :rinpatch_blurhash},
+          {:sizeable, "~>1.0 "}
```
Comment (mayel) [inline mix.exs]: we'd usually add that dependency to `deps.hex` file (only ones with special options, such as `optional: true` need to be included in mix.exs instead)

**Comment by mayel** [comment]
Thanks for taking the time! :) I added a couple comments but also happy to merge as is.

**Comment by chobeat** [comment]
have you completed the review? because the comments haven't appeared yet

**Review by mayel — COMMENTED** [review]
oops, didn't know the comments wouldn't appear

**Comment by mayel** [comment]
@chobeat I added support for ProcessTree in our Config module as discussed: https://github.com/bonfire-networks/bonfire_common/commit/0d5bfa1718a3455869209492f4584289eaebfa3e
so now it works async :)

**Comment by chobeat** [comment]
I saw your changes. In the way you rewrote the macro, doesn't this remove a default if present?

**Comment by mayel** [comment]
No because the value in Process (Process.put / ProcessTree.get) is only an override to the value in Config (Application env)

**Comment by chobeat** [comment]
I see. Do the new tests run successfully after your update on your machine? I think I recompiled all the new stuff but it's not really setting anything and the tests fail because the value is not modified. 

I did `git pull`, `just deps-get` then `just test extensions/bonfire_files/test/file_error_test.exs` and it recompiled, but the test fails. Is there any other step I should take to update the deps to include your change?

**Comment by mayel** [comment]
Yeah tests pass on my end. You probably need to update the bonfire_common dep (`just update-dep bonfire_common` or `just update` to update everything).

**Review by mayel — APPROVED** [review]
looks great, thanks!

**Pull request merged**